packages/services/src/competition.service.ts

```diff
@@ -2666,151 +2666,172 @@ export class CompetitionService {
   }
 
   /**
-   * Get competition rules for the active competition as a participant
-   * @param params Parameters for rules request
-   * @returns Formatted competition rules
+   * Builds initial balance descriptions from configuration
+   * @returns Array of formatted balance descriptions by chain
    */
-  async getCompetitionRules(params: {
-    agentId?: string;
-    isAdmin?: boolean;
-  }): Promise<CompetitionRulesData> {
-    try {
-      // Get active competition first, as rules are always for the active one
-      const activeCompetition = await this.getActiveCompetition();
-
-      if (!activeCompetition) {
-        throw new ApiError(
-          404,
-          "No active competition found to get rules for.",
-        );
+  private buildInitialBalanceDescriptions(): string[] {
```

Comment:
this is just breaking up the large getCompetitionRules function into multiple smaller helper functions